### PR TITLE
tool test: improving coverage, cleanups, minor fixes

### PR DESCRIPTION
**Pull request by ola-rozenfeld**
Includes https://github.com/bazelbuild/remote-apis-sdks/pull/473 because I couldn't figure out how to send a PR chain, sorry!

- adds a test for DownloadAction
- uses the `fakes.InputFile` whenever appropriate
- fixes the `ExecuteAction*` tests to not depend on implementation of the fake, and instead to mimics the user journey (downloading the inputs locally, then changing them).
- minor cleanups such as avoiding marshalling protos when unnecessarily, inlining small constants in tests, etc.

What I was actually doing is fixing https://github.com/bazelbuild/remote-apis-sdks/issues/460, but that PR got a bit too big with unrelated fixes and cleanups, so I decided to split these out.

Thank you!

### CODE REVIEW

File: go/pkg/fakes/server.go [hunk 1]
```diff
@@ -288,7 +288,7 @@ func (f *InputFile) apply(ac *repb.ActionResult, s *Server, execRoot string) err
 	if err := os.MkdirAll(filepath.Join(execRoot, filepath.Dir(f.Path)), os.ModePerm); err != nil {
 		return fmt.Errorf("failed to create input dir %v: %v", filepath.Dir(f.Path), err)
 	}
-	err := os.WriteFile(filepath.Join(execRoot, f.Path), nil, 0755)
+	err := os.WriteFile(filepath.Join(execRoot, f.Path), bytes, 0755)
```
Comment (ola-rozenfeld) [inline go/pkg/fakes/server.go:291]: That's from https://github.com/bazelbuild/remote-apis-sdks/pull/473

File: go/pkg/tool/tool_test.go [hunk 2]
```diff
@@ -83,8 +86,8 @@ Platform
 
 Inputs
 ======
-[Root directory digest: e23e10be0d14b5b2b1b7af32de78dea554a74df5bb22b31ae6c49583c1a8aa0e/75]
-a/b/input.txt: [File digest: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/0]
+[Root directory digest: 1b1d5b9e3eb97866805deca901299fcdde98b5747e147f35187e12c16e120186/75]
```
Comment (ola-rozenfeld) [inline go/pkg/tool/tool_test.go:89]: That's from https://github.com/bazelbuild/remote-apis-sdks/pull/473

File: go/pkg/tool/tool_test.go [hunk 3]
```diff
@@ -109,55 +112,98 @@ func TestTool_CheckDeterminism(t *testing.T) {
 	e, cleanup := fakes.NewTestEnv(t)
 	defer cleanup()
 	cmd := &command.Command{
-		Args:        []string{"foo bar baz"},
+		Args:        []string{"foo", "bar", "baz"},
```
Comment (ola-rozenfeld) [inline go/pkg/tool/tool_test.go:115]: This is here (and below) in order to not provide bad examples in our unit tests. E.g. "echo hello world" would not work as a single string; it needs to be split up.

File: go/pkg/tool/tool_test.go [hunk 4]
```diff
@@ -109,55 +112,98 @@ func TestTool_CheckDeterminism(t *testing.T) {
...
-	out := "output"
-	opt := &command.ExecutionOptions{AcceptCached: false, DownloadOutputs: true, DownloadOutErr: true}
-	_, acDg, _, _ := e.Set(cmd, opt, &command.Result{Status: command.SuccessResultStatus}, &fakes.OutputFile{Path: "a/b/out", Contents: out})
+	_, acDg, _, _ := e.Set(cmd, command.DefaultExecutionOptions(), &command.Result{Status: command.SuccessResultStatus}, &fakes.InputFile{Path: "i1", Contents: "i1"}, &fakes.InputFile{Path: "i2", Contents: "i2"}, &fakes.OutputFile{Path: "a/b/out", Contents: "out"})
 
 	client := &Client{GrpcClient: e.Client.GrpcClient}
 	if err := client.CheckDeterminism(context.Background(), acDg.String(), "", 2); err != nil {
 		t.Errorf("CheckDeterminism returned an error: %v", err)
 	}
-	// Now execute again with changed inputs.
```
Comment (ola-rozenfeld) [inline go/pkg/tool/tool_test.go:131]: This comment was a result of copy-paste; inputs do not actually change, only output does.

**Comment by ola-rozenfeld** [comment]
Preview of the Node Properties support change: https://github.com/ola-rozenfeld/remote-apis-sdks/pull/1

**Review by mrahs — APPROVED** [review]

**Pull request merged**